lang/py/avro/schema.py

```diff
@@ -521,6 +541,21 @@ def to_json(self, names=None):
         else:
             return self.props
 
+    def validate(self, datum):
+        """Return self if datum is a valid representation of this type of primitive schema, else None"""
+        is_valid = {
+            'null': lambda x: x is None,
+            'boolean': lambda x: isinstance(x, bool),
+            'string': lambda x: isinstance(x, unicode),
+            'bytes': lambda x: isinstance(x, bytes),
+            'int': lambda x: isinstance(x, int) and INT_MIN_VALUE <= x <= INT_MAX_VALUE,
+            'long': lambda x: isinstance(x, int) and LONG_MIN_VALUE <= x <= LONG_MAX_VALUE,
+            'float': lambda x: isinstance(x, (int, float)),
+            'double': lambda x: isinstance(x, (int, float)),
+        }.get(self.type, lambda x: False)(datum)
```

Comment:
Just having the primitive schema validation isolated from _all the validators_ is nice.

Reply:
I agree that this is not ideal. I'd prefer the types to exist outright, instead of `PrimitiveSchema`, just `IntSchema`, etc. Then mapping validators to types is natural. I've tried to do this several times, but got hung up in complexities. (There are a lot of Liskov violations in here.) IMO it's always worth another shot, but I wouldn't suggest taking it on in this PR, since this is already a big improvement.

I'd suggest making this dictionary a class attribute and then deciding the validator at init time. But other than that I think it's about as good as it can be without going too far off track.